pkg/controllers/customeradmin/group_controller.go

```diff
@@ -0,0 +1,108 @@
+package customeradmin
+
+import (
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	"github.com/openshift/openshift-azure/test/clients/openshift"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// https://docs.microsoft.com/en-us/graph/api/group-list
+// To list AAD groups, this code needs to have the clientID
+// of an application with the following permissions:
+// API: Microsoft Graph
+//   Application permissions:
+//      Read all groups
+
+const (
+	clientID     = "0d18f31e-2d82-45cf-8961-dbf2f62dba95" // jminter-groups-test
+	clientSecret = "Updy8eC6NTs9Xs4ltbk158F4BjzqHVYt1Dvj7Tk6pfc="
```

Comment:
I imagine this secret shouldn't be here.

Reply:
@jim-minter @asalkeld @kwoodson I have revoked the secret for 0d18f31e-2d82-45cf-8961-dbf2f62dba95 (jminter-groups-test) app registration in the portal.  You will need to regenerate one

Reply:
thanks @pweil- -- @asalkeld oops!

Reply:
Still not as good as my leak of the openshift-ci-robot token, though that was alleviated by the fact that github revokes access as soon as it detects a github token is pushed in its repos.

Reply:
sorry, i totally forgot that was there. Thanks for revoking the secret.